client/client.go

```diff
@@ -245,8 +245,13 @@ func (c *httpClusterClient) Do(ctx context.Context, act httpAction) (*http.Respo
 			continue
 		}
 		if resp.StatusCode/100 == 5 {
-			// TODO: make sure this is a no leader response
-			cerr.Errors = append(cerr.Errors, fmt.Errorf("client: etcd member %s has no leader", eps[k].String()))
+			switch resp.StatusCode {
+			case http.StatusInternalServerError:
+				// TODO: make sure this is a no leader response
+				cerr.Errors = append(cerr.Errors, fmt.Errorf("client: etcd member %s has no leader", eps[k].String()))
+			default:
+				cerr.Errors = append(cerr.Errors, fmt.Errorf("client: etcd member %s returns server error [%s] with body [%s]", eps[k].String(), http.StatusText(resp.StatusCode), body))
```

Comment:
1. ok

2. I check the no-leader response again, and find out that http.StatusServiceUnavailable could be returned at no-leader case too when GET /v2/members too. So it needs to be improved.

It is hard to fully resolve the TODO here because no-leader response cannot be recognized from others. It is always `{message: Internal Server Error}`. I could tackle this as follow-up.

What cases do you refer that we should not retry? I go through all server error HTTP status, and think all could retry. If some server has some internal error, other servers may work well. If it is a no-leader error, we could wait some time before retry.